components/applications-service/pkg/server/periodic.go

```diff
@@ -0,0 +1,252 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/chef/automate/components/applications-service/pkg/config"
+	"github.com/chef/automate/lib/cereal"
+	cerealRPC "github.com/chef/automate/lib/cereal/grpc"
+	"github.com/chef/automate/lib/cereal/patterns"
+	"github.com/chef/automate/lib/grpc/secureconn"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	rrule "github.com/teambition/rrule-go"
+)
+
+type JobScheduler struct {
+	CerealSvc *cereal.Manager
+}
+
+type DisconnectedServicesConfigV0 struct {
+	Enabled bool
+	Params  *DisconnectedServicesParamsV0
+}
+
+type DisconnectedServicesParamsV0 struct {
+	ThresholdDuration string `mapstructure:"threshold_duration"`
+}
+
+const (
+	cerealServiceMutualTLSName = "cereal-service"
+
+	DisconnectedServicesJobIntervalSeconds = 60
+	DisconnectedServicesJobName            = "disconnected_services"
+	DisconnectedServicesScheduleName       = "periodic_disconnected_services"
+)
+
+func ConnectToJobsManager(jobCfg *config.Jobs, connFactory *secureconn.Factory) (*cereal.Manager, error) {
+	svcURL := fmt.Sprintf("%s:%d", jobCfg.Host, jobCfg.Port)
+	conn, err := connFactory.Dial(cerealServiceMutualTLSName, svcURL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to connect to upstream job service %q at %q", cerealServiceMutualTLSName, svcURL)
+	}
+
+	be := cerealRPC.NewGrpcBackendFromConn("applications", conn)
+	//cerealSvc, err := cereal.NewManager(be)
+	cerealSvc, err := cereal.NewManager(be, cereal.WithTaskPollInterval(1*time.Second), cereal.WithTaskPollIntervalMaxJitter(100*time.Millisecond))
```

Comment:
curious why we need to run this more often

Reply:
In the tests I set the job to run once per second so I can verify that the system actually runs the task I expected it to run. In the studio on my machine this worked fine without this configuration, but in Ci the test always failed until I changed this. Since my job interval is 60s, the 1s poll interval doesn't seem too crazy.

Reply:
In the tests I set the job to run once per second so I can verify that the system actually runs the task I expected it to run. In the studio on my machine this worked fine without this configuration, but in Ci the test always failed until I changed this. Since my job interval is 60s, the 1s poll interval doesn't seem too crazy.

Reply:
Is this polling the cereal service if there is a task to run? Maybe it could be configurable too and run more frequently in CI.